### ISSUE TITLE
$Pscx:Preferences gets deleted when using Pscx.

### ISSUE THREAD

**Issue by Paebbels** [Pscx/Pscx#5]
When I use Pscx, I get these error messages on every `cd` usage after some time.

``` PowerShell
D:\git\PoC [paebbels/master = +1 ~0 -0 | +3 ~19 -0 !]> cd .\docs\
Cannot index into a null array.
At C:\Program Files\WindowsPowerShell\Modules\Pscx\3.2.1.0\Modules\CD\Pscx.CD.psm1:121 char:21
+                 if ($Pscx:Preferences['CD_GetChildItem'])
+                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [], RuntimeException
    + FullyQualifiedErrorId : NullArray
```

Before that happens, I executed a PowerShell script written by me, which loads Pscx (if not already loaded by the user) and uses `Invoke-BatchFile`. As far as I can see, my script uses only
- `Set-LocationEx` - `cd`
- `Invoke-BatchFile`

from Pscx 3.2.1.0. I already search in all `*.ps?1` files for write access to `$Pscx:` but I can't find any. I suspect it got deleted somewhere in the DLLs. I suspect, that the complete PSDrive Pscx vanished.

I also disabled `PoSh-Git`. It has no effect on this issue. My PowerShell version is 5.0 on Windows 7 x86-64.

**Here are some screens showing the problem:**

![pscx - error in cd](https://cloud.githubusercontent.com/assets/956109/16267347/9a8b3a78-3889-11e6-8e35-e1f393a71565.png)

![pscx - debug 1](https://cloud.githubusercontent.com/assets/956109/16267332/8a2d5df0-3889-11e6-843f-76170af317b1.PNG)

![pscx - debug 2](https://cloud.githubusercontent.com/assets/956109/16267336/904f8fb4-3889-11e6-82fe-dd811976d352.PNG)

![pscx - debug 3](https://cloud.githubusercontent.com/assets/956109/16267340/93b126c2-3889-11e6-8c1e-8b29bae3b4d1.PNG)

**My current very dirty work around is to unload and reload the Pscx module:**

``` PowerShell
rmo Pscx
ipmo Pscx
```

**Comment by rkeithhill** [comment]
That's strange that the PSDrive would go away.  This CD functionality, which I use all the time, probably should be broken out into its own module - say Pscx-CD (or PscxCD).  I've been wanting to do this for a while but have been spending most of my free time on [PowerShellEditorServices](https://github.com/powerShell/powershelleditorservices) and [Plaster](https://github.com/powerShell/plaster).

**Comment by Paebbels** [comment]
Oh sorry, I see my error messages are in German. shall I translate them?

I think the important message for `dir pscx:` is:

```
dir : No drive found. A drive with name "pscx" is not available.
```

I could overwrite the `cd` alias in my $PROFILE after importing Pscx, to reference the normal Set-Location CmdLet. The question is, who destroys `Pscx:` is it `cd` or `Invoke-BatchFile`?

I tried to offend `cd` by passing $null or other unusual values to it, but it always catches my false inputs and stays operational.

How does `Invoke-BatchFile` work?
Where can I find the code?

Regards
    Patrick

**Comment by rkeithhill** [comment]
I don't think it is `Invoke-BatchFile` because I use that all the time to configure my environment for visual studio tools.  But if you want to look it is [here](https://github.com/Pscx/Pscx/blob/master/Src/Pscx/Modules/Utility/Pscx.Utility.psm1#L745).

**Comment by Paebbels** [comment]
Oh, I see, the logic of `Invoke-BatchFile` is quite simple and should not effect `cd`.
